pkg/controller/storagecluster/storagecluster.go

```diff
@@ -278,6 +286,35 @@ func (c *Controller) validateSingleCluster(current *corev1.StorageCluster) error
 	return nil
 }
 
+// validateCustomAnnotations validate all custom annotations in the spec
+func (c *Controller) validateCustomAnnotations(current *corev1.StorageCluster) error {
+	if annotationMaps := current.Spec.Annotations; annotationMaps != nil {
+		for mapKey := range annotationMaps {
+			split := strings.Split(mapKey, "/")
+			if len(split) != 2 {
+				return fmt.Errorf("malformatted custom annotation locator: %s", mapKey)
+			}
+			kind := corev1.AnnotationKind(split[0])
+			if !kind.IsValid() {
+				return fmt.Errorf("%s: %s", util.InvalidCustomAnnotationKind, kind)
+			}
+			component := corev1.AnnotationComponent(split[1])
+			if !component.IsValid() {
+				return fmt.Errorf("%s: %s", util.InvalidCustomAnnotationComponent, component)
+			}
+			// TODO: OPERATOR-343 handle validation for different Portworx components and strict syntax validation
+			// So far custom annotation is only supported for Portworx storage pod
+			switch {
+			case kind == corev1.PodAnnotations && component == corev1.Portworx:
+			default:
```

Comment:
IMHO we don't need to be very strict on checking the supported annotations. It feels unnecessary. Rather just use the annotation if it matches a component, else ignore the extra annotations.
We have a lot of components and each component has several k8s objects. Regularly maintaining a list of these feels a bit redundant. Also, it spills the component details outside of the component package.